internal/decoder/string.go

```diff
@@ -270,6 +270,7 @@ func stringBytes(s *Stream) ([]byte, error) {
 			r, size := utf8.DecodeRune(s.buf[cursor:])
 			if r == utf8.RuneError {
 				s.buf = append(append(append([]byte{}, s.buf[:cursor]...), runeErrBytes...), s.buf[cursor+1:]...)
+				s.bufSize = int64(len(s.buf))
```

Comment:
@vadimInshakov  I would like some more explanation as to why the bufSize needs to be changed at this place

Reply:
We get panic here https://github.com/goccy/go-json/blob/master/internal/decoder/stream.go#L201

Panic occurs because the cursor is pointing out of bounds s.buf. 
s.buf slice boundaries are set here: https://github.com/goccy/go-json/blob/master/internal/decoder/stream.go#L195)

But these boundaries do not match where the cursor is pointing. My change makes it so that the cursor does not go beyond the buffer bounds (on s.buf resize we change bufSize accordingly).